vars/buildPlugin.groovy

```diff
@@ -139,7 +139,25 @@ def call(Map params = [:]) {
                   mavenOptions += 'clean install'
                   try {
                     boolean usingArtifactCachingProxy = false
-                    if (artifactCachingProxyEnabled) {
+
+                    // If the build concerns a pull request, check if there is "skip-artifact-caching-proxy" label applied in case the user doesn't wan't ACP
+                    boolean prLabelsContainSkipACP = false
+                    final String skipACPLabel = 'skip-artifact-caching-proxy'
+                    if (!env.BRANCH_IS_PRIMARY) {
+                      withCredentials([
+                        usernamePassword(credentialsId: 'github-app-infra', usernameVariable: 'GITHUB_APP', passwordVariable: 'GH_TOKEN')
+                      ]) {
+                        if (isUnix()) {
+                          prLabelsContainSkipACP = sh(script: 'gh pr view $CHANGE_URL --json labels | grep --ignore-case \'"skip-artifact-caching-proxy"\'', returnStatus: true) == 0
+                        } else {
+                          prLabelsContainSkipACP = bat(script: 'gh pr view $CHANGE_URL --json labels | findstr /i \'"skip-artifact-caching-proxy"\'', returnStatus: true) == 0
```

Comment:
Alas not all of our Windows Agent template have `gh` installed:

- ✅  Windows VMs (Azure and AWS) already have it (https://github.com/jenkins-infra/packer-images/blob/5178d7d49d9f087a9d6adf05241f336c769c9fbf/provisioning/windows-provision.ps1#L243-L255)
- ❌ We need to install it on the Windows Container agents though:
  - JDK11: https://github.com/jenkins-infra/docker-inbound-agents/blob/main/maven/jdk11/Dockerfile.nanoserver
  - JDK8: https://github.com/jenkins-infra/docker-inbound-agents/blob/main/maven/jdk8/Dockerfile.nanoserver
  - JDK17: https://github.com/jenkins-infra/docker-inbound-agents/blob/main/maven/jdk17/Dockerfile.nanoserver
  - JDK19: https://github.com/jenkins-infra/docker-inbound-agents/blob/main/maven/jdk19/Dockerfile.nanoserver

otherwise this command will fail

Reply:
I'm thinking about replacing the `gh` command by a `curl` call, something like 
```
curl \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer $GITHUB_TOKEN"\
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/jenkinsci/jenkins-infra-test-plugin/issues/57/labels 
```
followed by the same kind of grep.

WDYT @dduportal?

Reply:
Good idea if it works out of the box on Windows \o/

Reply:
(but adding `gh` is not complicated either: as soon as it's merged, it will be immediatly pushed and deployed as the windows container are using `latest`)

pick your choise, both are fine 👍